src/main/java/com/microsoft/aad/msal4j/DeviceCodeFlowRequest.java

```diff
@@ -41,10 +41,17 @@ DeviceCode acquireDeviceCode(String url,
                                  Map<String, String> clientDataHeaders,
                                  ServiceBundle serviceBundle) throws Exception {
 
-        String urlWithQueryParams = createQueryParamsAndAppendToURL(url, clientId);
```

Comment:
this function is not used any more? should it be deleted ?
it might make sense to keep the code which creates query parameters incapsulated in this function

Reply:
In the latest commit I refactored the method from "createQueryParamsAndAppendToURL" to just "createQueryParams", and moved all of the logic for setting up the query parameters (which go into the request body now instead of URL) back into that method.